draft-frindell-webtrans-devious-baton.md

```diff
@@ -183,6 +193,12 @@ single datagram.
 
 ## Session Closure
 
+Each endpoint tracks the number of active batons.  It is initally equal to the
+client's `count` parameter.  Each time a baton exchange completes or is reset,
+the number of active batons is decreased by 1. When the number of active batons
+reaches 0 and at least one baton exchange terminated when the baton reached 0,
+the endpoint MUST close the WebTransport session with no error.
+
```

Comment:
I realized this is a bit vague -- only an endpoint that reaches an active count of 0 when receiving a message or RESET_STREAM should close the session.  Even with that restriction, there's still a possible race condition where both endpoints reach 0 simultaneously as a result of received messages.

Thoughts?

Also curious if it's worth maintaining the distinction between close with no error and GAME_OVER.

Reply:
My initial implementation was just closing everything on first baton to reach zero, but I like the countdown better -- more deterministic when running tests.

The spec says that "0" can be sent at any moment. So it should be fine if the client has seen 15 turns of the baton already, decides that's enough, and sends the next baton as "0". So it is not just reset.

I dislike assigning special meaning to reset, because of the inherent race condition between "receive FIN" and "receive RESET".